src/metabase/db/custom_migrations.clj

```diff
@@ -290,8 +304,22 @@
                             (t2/reducible-query {:select [:id :visualization_settings :result_metadata]
                                                  :from   [:report_card]
                                                  :where  [:and
-                                                          [:<> :result_metadata nil]
-                                                          [:<> :result_metadata "[]"]]})))))
+                                                          [:like :visualization_settings "%column_settings%"]
+                                                          [:like :result_metadata "%join-alias%"]]}))))
```

Comment:
I updated this where clauses because this is a performance optimization that I hadn't considered before. There will be many cards without `:join-alias` in the result_metadata or `visualization_settings.column_settings`. We can skip extracting and parsing JSON for these cards.